commands/show.go

```diff
@@ -90,4 +97,5 @@ var showCmd = &cobra.Command{
 
 func init() {
 	RootCmd.AddCommand(showCmd)
+	showCmd.Flags().BoolVarP(&showId,"id","i",false,"Select field to display: id")
```

Comment:
While I completely agree that it's a good feature to have (I had that in mind already), I'm not a fan of the current UX:

- `id` refers to the full 40 character long identifier. "Human ID" is more an internal naming from the geek world, so let's call that "short id"
- it should be open to more field than just the id. There is at least
  - id
  - short id
  - status
  - title
  - last comment
  - last edit time (regular time)
  - last edit time (lamport time)
  - labels (one by line)
  - author's name
  - author's email

So ...

What about a unique `--field=` to select what field to display ? This could very well combine with a future `--enc=` flage to select what kind of output we want (normal, json, ...). In the future, a multi valued `--fields=` could be added to select many fields at the same time as well.

There is an example of how to deal with these flags in the `ls` command.

Reply:
Yeah, leave the json thing for later, it needs extensive plumbing and a battle plan on its own.

I wrote an overview of the data model here: https://github.com/MichaelMure/git-bug/blob/master/doc/model.md

There is also the comment, either directly in the code or through https://godoc.org/github.com/MichaelMure/git-bug

But in short:

- `bug.Bug` hold the raw data, that is the list of edit operations for a bug
- `bug.Snapshot` hold a compiled version for the various UI to use
- `bug.WithSnapshot` is a special purpose `Bug` that maintain a `Snapshot` up to date incrementally, each time a new operation is added, for performance
- `bug.Interface` is an interface to manipulate `Bug` or `WithSnapshot` transparently